pywebhdfs/webhdfs.py

```diff
@@ -132,8 +147,17 @@ def append_file(self, path, file_data, **kwargs):
 
         # make the initial APPEND call to the HDFS namenode
         optional_args = kwargs
-        uri = self._create_uri(path, operations.APPEND, **optional_args)
-        init_response = requests.post(uri, allow_redirects=False)
+
+        hosts = self._resolve_federation(path)
+        for host in hosts:
+            uri = self._create_uri(host, path, operations.APPEND,
```

Comment:
the code in the for loop is repetitive in nature for all these methods.  Maybe there is a way to do some dynamic foo alleviate that?  We don't necessarily have to do that for this PR.